### ISSUE TITLE
Tracing integrations: allow for custom trace/run_ids

### ISSUE THREAD

**Issue by marcklingen** [deepset-ai/haystack-core-integrations#1264]
**Is your feature request related to a problem? Please describe.**

When working on LLM applications and tracing them, the trace_id is usually key to later on:
- add additional non-haystack spans to the trace
- record user feedback or other evaluation metrics for the trace
- be able to fetch the single trace via API at a later point
- use distributed tracing to log to the same trace id

Currently, there is no way of customizing the trace_id (at least not when using the Langfuse intgeration).

**Describe the solution you'd like**

I like the DX of LangChain where a `run_id` can be set via the run_config and is later on used as a trace_id for tracing. See example [here](https://langfuse.com/docs/integrations/langchain/tracing#predefined-trace-id--add-evaluation-or-user-feedback-score).

I am unsure where this is best set within Haystack, probably as part of the `run` method.

**Comment by vblagoje** [comment]
@marcklingen can't we simply use invocation_context in [LangfuseConnector](https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/langfuse/src/haystack_integrations/components/connectors/langfuse/langfuse_connector.py#L163) and then perhaps we can access these values user provided in run to convert them to tags/markers or whatever is needed to mark the trace?

**Comment by vblagoje** [comment]
Unless I'm overlooking something  one should be able to achieve this by adjusting DefaultSpanHandler from https://github.com/deepset-ai/haystack-core-integrations/pull/1313 or perhaps implement a custom handler class and leave it here for others to see as an example of customization

**Comment by sjrl** [comment]
Hey @marcklingen so this is possible with the current Langfuse Integration. It's not as succinct but you can do the following
```python
from haystack_integrations.tracing.langfuse.tracer import tracing_context_var

new_ctx = current_ctx.copy()
new_ctx["trace_id"] = "custom-trace-123"
new_ctx["user_id"] = "user-456"
new_ctx["tags"] = ["production"]

# Set the new context
token = tracing_context_var.set(new_ctx)

response = pipe.run(...)

# Optionally reset the context later
tracing_context_var.reset(token)
```

This works for the following variables:
- "trace_id"
- "user_id"
- "session_id"
- "tags"
- "version"

We can see about adding a convenience function that helps make it easier to understand what can be passed. Perhaps something like 
```python
def set_langfuse_context(trace_id=None, user_id=None, session_id=None, tags=None, version=None):
    current_ctx = tracing_context_var.get({})
    new_ctx = current_ctx.copy()
    if trace_id is not None:
        new_ctx["trace_id"] = trace_id
    if user_id is not None:
        new_ctx["user_id"] = user_id
    if session_id is not None:
        new_ctx["session_id"] = session_id
    if tags is not None:
        new_ctx["tags"] = tags
    if version is not None:
        new_ctx["version"] = version
    return tracing_context_var.set(new_ctx)

def reset_langfuse_context(token=None):
    if token:
        tracing_context_var.reset(token)
    else:
        tracing_context_var.set({})
```

Would this cover the use case you had in mind?

**Comment by Hansehart** [comment]
Yes, this works with the current Langfuse Integration! Here's a simple example:

```
from haystack_integrations.tracing.langfuse.tracer import tracing_context_var

[...]

# Set custom trace context
current_ctx = tracing_context_var.get({})
new_ctx = current_ctx.copy()
new_ctx["trace_id"] = "custom-trace-123"
new_ctx["user_id"] = "user-456"
token = tracing_context_var.set(new_ctx)

# Run your pipeline - it will use the custom trace_id
response = pipe.run({"query": "What is Haystack?"})

# Reset context
tracing_context_var.reset(token)
```
